File: doc/preprocessing.R
# install.packages("dplyr")
# install.packages("ggmap")
library(dplyr)
library(ggmap)

data <- read.csv("data/DisasterDeclarationsSummaries.csv")

register_google("AIzaSyDLtO6ZqLGYMNlOYHi-NwrH146ySmW3qR0")

# Create a new data frame with latitude and longitude columns
geocoded_data <- data %>%
  rowwise() %>%
  mutate(location = paste(fipsStateCode, fipsCountyCode, designatedArea, state, sep = ", "),
         geo_data = geocode(location),
         latitude = ifelse(!is.na(geo_data$lat), geo_data$lat, NA),
         longitude = ifelse(!is.na(geo_data$lon), geo_data$lon, NA)) %>%
  select(-location, -geo_data)

# Extract year and month from the declarationDate column
geocoded_data <- geocoded_data %>%
  mutate(declarationYear = as.integer(format(as.POSIXct(declarationDate, format = "%Y-%m-%dT%H:%M:%S.000Z"), "%Y")),
         declarationMonth = as.integer(format(as.POSIXct(declarationDate, format = "%Y-%m-%dT%H:%M:%S.000Z"), "%m")))

# View the updated data frame
head(geocoded_data)

write.csv(geocoded_data, "out/DisasterDeclarationsSummariesCleaned.csv", row.names = FALSE)

File: doc/preprocessing.py
import numpy as np
import pandas as pd

src = pd.read_csv("../data/us_county_latlng.csv")
src['fipsCountyCode'] = src['fipsCountyCode'].apply(lambda x: f"{x:05}")

df = pd.read_csv("../data/DisasterDeclarationsSummaries.csv")
df['declarationYear'] = df['declarationDate'].str.extract(r'(\d{4})')
df['latitude'] = np.nan
df['longitude'] = np.nan

for index, row in df.iterrows():
    string_number = f"{df.at[index,'fipsStateCode']:02}{df.at[index,'fipsCountyCode']:03}"
    matching_rows = src.loc[src['fipsCountyCode'] == string_number]
    if not matching_rows.empty:
        df.at[index, 'latitude'] = matching_rows.iloc[0]['lat']
        df.at[index, 'longitude'] = matching_rows.iloc[0]['lng']
df.to_csv("../out/DisasterDeclarationsSummariesCleaned.csv")

incidentTypes = df['incidentType'].unique()
incidentTypesCleaned = [s.replace(" ", "").replace("/", "") for s in incidentTypes]
for i in range(len(incidentTypes)):
    temp = df.loc[df['incidentType']==incidentTypes[i]]
    temp.to_csv(f"../out/DisasterDeclarationsSummaries{incidentTypesCleaned[i]}.csv", index=False)

for declarationYear in df['declarationYear'].unique():
    temp = df.loc[df['declarationYear']==declarationYear]
    temp.to_csv(f"../out/DisasterDeclarationsSummaries{declarationYear}.csv", index=False)